services/destinationService.js

```diff
@@ -17,3 +19,81 @@ export async function getAllDestinations() {
 
   return sanitizedDestinations;
 }
+
+export async function getDestinationById(id) {
+  await dbConnect();
+
+  const destination = await Destination.findOne({ _id: id });
+  const sanitizedDestination = {
+    id: destination.id,
+    name: destination.name,
+    startDate: destination.startDate,
+    endDate: destination.endDate,
+    hotel: destination.hotel,
+    transport: destination.transport,
+    tripId: destination.tripId,
+  };
+
+  return sanitizedDestination;
+}
+
+export async function getDestinationsByTripId(id) {
+  await dbConnect();
+  const destinations = await Destination.find({ tripId: id });
+  if (!Array.isArray(destinations)) return { status: 500, error: "Internal server error" };
+
+  const sanitizedDestinations = destinations.map((destination) => ({
+    id: destination.id,
+    name: destination.name,
+    startDate: destination.startDate,
+    endDate: destination.endDate,
+    hotel: destination.hotel,
+    transport: destination.transport,
+    tripId: destination.tripId,
+  }));
+
+  return sanitizedDestinations;
+}
+
+export async function createDestination(body, tripId) {
+  if (!validateId(tripId)) return { status: 400, error: "Invalid tripId" };
```

Comment:
It doesn't really make the most sense to check whether the tripId has the correct format. I would not check for that at all. The only thing worth checking is if the trip with the respective id exists in the database. Like this you can create a destination with a correctly formatted tripId without any trip matching the tripId => basically dead weight in your database.

Reply:
Changed validity check to database request if id exists